include/boost/random/detail/uniform_int_float.hpp

```diff
@@ -60,7 +60,7 @@ class uniform_int_float
     result_type operator()()
     {
         base_result range = static_cast<base_result>((max)())+1;
-        return static_cast<result_type>(_rng() * range);
+        return static_cast<result_type>(static_cast<base_result>(_rng() * range));
```

Comment:
AMDG

On 03/12/2015 12:00 PM, jzmaddock wrote:
>> @@ -60,7 +60,7 @@ class uniform_int_float
>>      result_type operator()()
>>      {
>>          base_result range = static_cast<base_result>((max)())+1;
>> -        return static_cast<result_type>(_rng() * range);
>> +        return static_cast<result_type>(static_cast<base_result>(_rng() * range));
> 
> Fixed upstream in multiprecision and reverted this change: unfortunately adding explicit conversion operators to the expression templates caused new issues in chi_squared_distribution.hpp which required the "obvious" casts.  You win some you lose some I guess... whatever adding the explicit conversion operators to the expression templates is probably the right thing to do anyway.
> 

Did you push this update?  I don't see it on my end.

Also:
- There are several places that change to 3 space indent,
  when the surrounding code is all indented to 4 spaces.
- traits.hpp has a *DETAIL* #include guard.
- doc/Jamfile.v2 refers to detail/traits.hpp
- In independent_bits_engine, I'd prefer all the
  private functions to be at the end.  base_result_type
  should be hidden from doxygen (and I'd also prefer
  not to rename it).

In Christ,
Steven Watanabe